python/delete_unlinked_top_containers.py

```diff
@@ -0,0 +1,62 @@
+import asnake.logging as logging
+from asnake.client import ASnakeClient
+import argparse
+
+logging.setup_logging(filename="archivessnake.log", level="INFO")
+# set label for custom logger - all output will be in archivessnake.log
+logger = logging.get_logger("delete_unlinked_top_containers")
+client = ASnakeClient()
+
+
+def container_is_unlinked(top_container_uri: str) -> bool:
+    top_container = client.get(top_container_uri).json()
+    if len(top_container["collection"]) == 0:
+        return True
+    return False
+
+
+def container_exists(top_container_uri: str) -> bool:
+    top_container = client.get(top_container_uri).json()
+    if "error" in top_container.keys():
+        logger.error(
+            f"Error retrieving top container {top_container_uri}: {top_container['error']}"
+        )
+        return False
+    return True
+
+
+def delete_unlinked_top_containers(container_list_file: str):
+    logger.info(f"Reading top container URIs to delete from {container_list_file}")
+    with open(container_list_file, "r") as f:
+        container_list = f.readlines()
+    container_list = [x.strip() for x in container_list]
+
+    deleted_count = 0
+    skipped_uri_list = []
+    for container in container_list:
+        if not container_exists(container):
+            # error message already logged in container_exists()
+            skipped_uri_list.append(container)
+            continue
+        if container_is_unlinked(container):
+            logger.info(f"Deleting unlinked top container: {container}")
+            client.delete(container)
```

Comment:
What happens if this is run by a user who does not have permission to delete containers?  For example, our default accounts in production just have `view_repository`, I think.  This is unlikely to happen, and we haven't accounted for this in Alma API scripts... but we should start.

If the API or client here raise an exception, we should decide what to do.  I'm OK with not catching it and crashing immediately, if the error message is meaningful.  For example, if this program is run with a non-existing/unreadable file, the immediate `FileNotFoundError` is fine - the error is clear.

If the exception error message is not really clear about not having permission to delete, we should catch, log (and possibly also print) a useful message, and immediately exit.

If there's no exception, just an error embedded in a response similar to the payload of "container not found", and the program would therefore try and fail to delete every URI in the file... we should detect the problem, raise/create and log/print a meaningful exception of our own, and immediately exit.